scheduler/src/test/java/gov/nasa/jpl/aerie/scheduler/UncontrollableDurationTest.java

```diff
@@ -82,12 +85,11 @@ public void testNonLinear(){
     final var solver = new PrioritySolver(problem);
     final var plan = solver.getNextSolution().get();
     solver.printEvaluation();
-    assertTrue(TestUtility.containsActivity(plan, planningHorizon.fromStart("PT1M51S"), planningHorizon.fromStart("PT6M51S"), problem.getActivityType("SolarPanelNonLinear")));
-    assertTrue(TestUtility.containsActivity(plan, planningHorizon.fromStart("PT35M11S"), planningHorizon.fromStart("PT40M11S"), problem.getActivityType("SolarPanelNonLinear")));
-    assertTrue(TestUtility.containsActivity(plan, planningHorizon.fromStart("PT18M31S"), planningHorizon.fromStart("PT23M31S"), problem.getActivityType("SolarPanelNonLinear")));
-    assertTrue(TestUtility.containsActivity(plan, planningHorizon.fromStart("PT23M31S"), planningHorizon.fromStart("PT25M"), problem.getActivityType("SolarPanelNonLinear")));
-    assertTrue(TestUtility.containsActivity(plan, planningHorizon.fromStart("PT6M51S"), planningHorizon.fromStart("PT8M20S"), problem.getActivityType("SolarPanelNonLinear")));
-    assertTrue(TestUtility.containsActivity(plan, planningHorizon.fromStart("PT40M11S"), planningHorizon.fromStart("PT41M40S"), problem.getActivityType("SolarPanelNonLinear")));
+    assertTrue(TestUtility.containsActivity(plan, planningHorizon.fromStart("PT11M40S"), planningHorizon.fromStart("PT16M40S"), problem.getActivityType("SolarPanelNonLinear")));
+    assertTrue(TestUtility.containsActivity(plan, planningHorizon.fromStart("PT28M20S"), planningHorizon.fromStart("PT33M20S"), problem.getActivityType("SolarPanelNonLinear")));
+    assertTrue(TestUtility.containsActivity(plan, planningHorizon.fromStart("PT0S"), planningHorizon.fromStart("PT1M29S"), problem.getActivityType("SolarPanelNonLinear")));
+    assertTrue(TestUtility.containsActivity(plan, planningHorizon.fromStart("PT16M40S"), planningHorizon.fromStart("PT18M9S"), problem.getActivityType("SolarPanelNonLinear")));
+    assertTrue(TestUtility.containsActivity(plan, planningHorizon.fromStart("PT33M20S"), planningHorizon.fromStart("PT34M49S"), problem.getActivityType("SolarPanelNonLinear")));
```

Comment:
Because the initial bounds are now checked before iteration, the timings of activities have changed.